ingestfile

```diff
@@ -68,28 +62,171 @@ _remove_mediaid_from_queue(){
     mv "${QUEUEFILE}_tmp" "${QUEUEFILE}"
 }
 
+_blackframetest(){
+    _black_at_ends "${INPUT}"
+    if [ "${HEAD_BLACK}" -gt "30" ] ; then
+        _report -wt "WARNING - There are at least ${HEAD_BLACK} frames of black at the beginning of ${INPUT}"
+        _report -qn "Enter q to quit, any other key to continue: "
+        read A1
+        [ "${A1}" == "q" ] && exit 0
+        _ask_intime
+    fi
+    if [ "${TAIL_BLACK}" -gt "30" ] ; then
+        _report -wt "WARNING - There are at least ${TAIL_BLACK} frames of black at the end of ${INPUT}"
+        _report -qn "Enter q to quit, any other key to continue: "
+        read A2
+        [ "${A2}" == "q" ] && exit 0
+        _ask_outtime
+    fi
+}
+
+_phasechecktest(){
+    _report -dt "Checking audio phase for $(basename "${INPUT}")"
+    _get_phase_warning "${INPUT}"
+    if [ "${PHASE_ERROR}" -gt "0" ] ; then
+        _report -w "WARNING - There are ${PHASE_ERROR} sections of out-of-phase audio"
+        _report -qn "Enter q to quit, any other key to continue: "
+        read A3
+        [ "${A3}" == "q" ] && exit 0
+    fi
+}
+
+_makebroadcastcopies(){
+    "${SCRIPTDIR}/makederiv" -T broadcast "${MAKEDERIVOPTS[@]}" "${OUTDIR_INGESTFILE}/${MEDIAID}"
+    if [ "${SLATE}" == "Yes" ] ; then
+        "${SCRIPTDIR}/makederiv" -T broadcast -s "${MAKEDERIVOPTS[@]}" "${OUTDIR_INGESTFILE}/${MEDIAID}"
+    fi
+    if [ "${DELIVERYTOOMNEON}" == "Y" ] ; then
+        _delivertoomneon
```

Comment:
curious if this would work, since this function is defined later in the document